module/zfs/zfs_ioctl.c

```diff
@@ -7479,11 +7480,22 @@ zfsdev_ioctl_common(uint_t vecnum, zfs_cmd_t *zc, int flag)
 		goto out;
 
 	/* legacy ioctls can modify zc_name */
-	saved_poolname = kmem_strdup(zc->zc_name);
+	/*
+	 * Can't use kmem_strup() as we might truncate the string and
+	 * kmem_strfree() would then free with incorrect size.
+	 * Further complicated by that it is used with tsd_set() and
+	 * freed over in zfs_ioc_log_history(kmem_strfree()) so the
+	 * size of saved_poolname has to be the exact truncated version
```

Comment:
how about something like: `The string is freed by calling kmem_strfree() in zfs_ioc_log_history(), so the allocation needs to be for the exact truncated size.`

Reply:
It was more my thought process at the time "oh wait, I can't just save the len, as its given to tsd_set() and I went and changed it to tsd_set(kmem_strdup()) - comment serves no point now, I'll remove it